coap-gateway/service/refreshToken.go

```diff
@@ -89,11 +89,21 @@ func refreshTokenPostHandler(req *mux.Message, client *Client) {
 		return
 	}
 
-	token, err := client.server.provider.Refresh(req.Context, refreshToken.RefreshToken)
+	var token *oauth2.Token
+	for _, provider := range client.server.providers {
+		token, err = provider.Refresh(req.Context, refreshToken.RefreshToken)
```

Comment:
maybe a log if `err != nil` or is it expected behavior for only one provider to succeed?

Reply:
we don't know which provider is needed to use so we try all of them